Server/API/rooms/getRoomById.go

```diff
@@ -0,0 +1,28 @@
+package rooms
+
+import (
+	"context"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/labstack/echo/v4"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"ishtaloo.io/DB/Collections"
+	entities "ishtaloo.io/Entities"
+)
+
+func getRoomById(c echo.Context) error {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	roomsCollection := Collections.RoomsGetCollection()
+	roomId := c.Param("roomId")
+	id, _ := primitive.ObjectIDFromHex(roomId)
+	var room entities.Room
+	if err := roomsCollection.FindOne(ctx, bson.M{"_id": id}).Decode(&room); err != nil {
```

Comment:
Why is err  inside the if block? Shouldn't the if be checking if the err is nil?

Reply:
that the syntax they used in the mongo toturial, it still works.

Reply:
Cool